### PR TITLE
fix(code-server): change code-server source from forked repo to upstr…

### PR DESCRIPTION
**Pull request by Souheil-Yazji**
# Description

- Points the Jupyterlab and sas images to the https://github.com/coder/code-server repo instead of the Statcan fork.
- Updates code-server from v3.10.2(vs-Code v1.56.1) -> v4.5.1(vs-Code v1.68.1)
- Includes[ feature from upstream](https://github.com/coder/code-server/pull/5055) to remove the download button in continuity with:

  - https://github.com/StatCan/daaas/issues/568 
  - https://github.com/StatCan/daaas/issues/457

### Note
The ability to disable file downloads is done through a cli flag which can be set using an environment variable. I noticed that users can modify environment variables on their jupyterlab notebooks, so I went ahead and tried to modify the variable `CS_DISABLE_FILE_DOWNLOADS = 0` but I still couldn't see the "download" option in vs-code. **Double check this during review**. 

## Impact
Closes https://github.com/StatCan/daaas/issues/1100
Partially resolves https://github.com/StatCan/aaw-kubeflow-containers/issues/337
Deprecates https://github.com/StatCan/code-server

# Tests / Quality Checks

## Automated Testing/build and deployment
- [ ] Does the image pass CI successfully (build, pass vulnerability scan, and pass automated test suite)?
- [ ] If new features are added (new image, new binary, etc), have new automated tests been added to cover these?
- [x] If new features are added that require in-cluster testing (e.g. a new feature that needs to interact with kubernetes), have you added the `auto-deploy` tag to the PR before pushing in order to build and push the image to ACR so you can test it in cluster as a custom image?

## VS Code tests

- [x] Does VS Code open?
- [x] Can you install extensions?

## Code review

- [x] Have you added the `auto-deploy` tag to your PR before your most recent push to this repo?  This causes CI to build the image and push to our ACR, letting reviewers access the built image without having to create it themselves
- [x] Have you chosen a reviewer, attached them as a reviewer to this PR, and messaged them with the SHA-pinned image name for the final image to test on the **dev cluster** (e.g. `k8scc01covidacrdev.azurecr.io/jupyterlab-cpu:746d058e2f37e004da5ca483d121bfb9e0545f2b`)?

### CODE REVIEW

File: docker-bits/6_jupyterlab.Dockerfile [hunk 1]
```diff
@@ -11,27 +11,30 @@
...
     && rm -f /etc/apt/sources.list.d/vscode.list \
-    && mkdir -p /etc/share/code-server/extensions
+    && mkdir -p $XDG_DATA_HOME/code-server/extensions 
 
 # Fix for VSCode extensions and CORS
-ENV XDG_DATA_HOME=/etc/share
-ENV SERVICE_URL=https://extensions.coder.com/api
-COPY vscode-overrides.json $XDG_DATA_HOME/code-server/User/settings.json
-ARG SHA256py=d32d8737858661451705faa9f176f8a1a03485b2d9984de40d45cc0403a3bcf4
+# COPY vscode-overrides.json $XDG_DATA_HOME/code-server/User/settings.json
```
Comment (vexingly) [inline docker-bits/6_jupyterlab.Dockerfile:34]: Unless you're still testing this solution, all the commented out older code should be removed :)

**Review by vexingly — APPROVED** [review]

File: docker-bits/6_jupyterlab.Dockerfile (hunk 1 above)
Comment (Souheil-Yazji) [reply docker-bits/6_jupyterlab.Dockerfile]: resolved in 5b94ec1

**Review by vexingly — APPROVED** [review]
LGTM!

**Comment by Souheil-Yazji** [comment]
Image: k8scc01covidacrdev.azurecr.io/jupyterlab-cpu:997b7919ef76c777b7caaafa89cbf8dbd25ff78c

**Pull request merged**